json.c

```diff
@@ -202,810 +216,914 @@ static int new_value (json_state * state,
 {
-   json_value * cur_value;
+    json_value * cur_value;
 
-   if (!value)
-      return;
+    if (!value)
+        return;
 
-   value->parent = 0;
+    value->parent = 0;
 
-   while (value)
-   {
-      switch (value->type)
-      {
-         case json_array:
+    while (value)
+    {
+        switch (value->type)
+        {
+            case json_array:
 
-            if (!value->u.array.length)
-            {
-               settings->mem_free (value->u.array.values, settings->user_data);
-               break;
-            }
+                if (!value->u.array.length)
+                {
+                    settings->mem_free (value->u.array.values, settings->user_data);
+                    break;
+                }
 
-            value = value->u.array.values [-- value->u.array.length];
-            continue;
+                value = value->u.array.values [-- value->u.array.length];
+                continue;
 
-         case json_object:
+            case json_object:
 
-            if (!value->u.object.length)
-            {
-               settings->mem_free (value->u.object.values, settings->user_data);
-               break;
-            }
+                if (!value->u.object.length)
+                {
+                    settings->mem_free (value->u.object.values, settings->user_data);
+                    break;
+                }
 
-            value = value->u.object.values [-- value->u.object.length].value;
-            continue;
+                value = value->u.object.values [-- value->u.object.length].value;
+                continue;
 
-         case json_string:
+            case json_string:
 
-            settings->mem_free (value->u.string.ptr, settings->user_data);
-            break;
+                settings->mem_free (value->u.string.ptr, settings->user_data);
+                break;
 
-         default:
-            break;
-      };
+            default:
+                break;
+        };
 
-      cur_value = value;
-      value = value->parent;
-      settings->mem_free (cur_value, settings->user_data);
-   }
+        cur_value = value;
+        value = value->parent;
+        settings->mem_free (cur_value, settings->user_data);
+    }
 }
 
 void json_value_free (json_value * value)
 {
-   json_settings settings = { 0 };
-   settings.mem_free = default_free;
-   json_value_free_ex (&settings, value);
+    json_settings settings = { 0 };
+    settings.mem_free = default_free;
+    json_value_free_ex (&settings, value);
 }
 
+int trailing_garbage (const json_char * ptr)
+{
+    json_char marker = *ptr;
+    do
+    {
+        ptr--;
+    }
+    while (isspace(*ptr));
+
+    switch (*ptr)
+    {
+        case '}':
+        case '{':
+        case ']':
+        case '[':
+        case '"':
+            return 0;
+
+        case 'e':
+            marker = *(--ptr);
+            if (marker == 's')
+            {
+                if (*(--ptr) == 'l' && *(--ptr) == 'a' && *(--ptr) == 'f')
```

Comment:
Here you decrement `--ptr` multiple times. Are there cases where you could decrement pas the beginning of the buffer?